### PR TITLE
feat: refactor PIA and CAA remove procedures

### PR DESCRIPTION
**Pull request by shuffledex**

### CODE REVIEW

File: src/api/procedures/removeCorporateActionsAgent.ts [hunk 1]
```diff
@@ -30,19 +32,20 @@ export async function prepareRemoveCorporateActionsAgent(
...
-  ]);
+  const agents = await securityToken.corporateActions.getAgents();
 
-  if (owner.did === agent[0].did) {
+  if (agents.length !== 1) {
     throw new PolymeshError({
       code: ErrorCode.ValidationError,
-      message: 'There is no set Corporate Actions Agent',
+      message:
+        'We can not perform this procedure with more than one Corporate Actions Agent involved',
```
Comment (monitz87) [inline src/api/procedures/removeCorporateActionsAgent.ts:41]: "There must be one (and only one) Corporate Actions Agent assigned to this Security Token"

File: src/api/procedures/removePrimaryIssuanceAgent.ts [hunk 2]
```diff
@@ -1,59 +1,72 @@
...
+
+  const securityToken = new SecurityToken({ ticker }, context);
+
+  const { primaryIssuanceAgents } = await securityToken.details();
+
+  if (primaryIssuanceAgents.length !== 1) {
+    throw new PolymeshError({
+      code: ErrorCode.ValidationError,
+      message:
+        'We can not perform this procedure with more than one Primary Issuance Agent involved',
```
Comment (monitz87) [inline src/api/procedures/removePrimaryIssuanceAgent.ts:41]: "There must be one (and only one) Primary Issuance Agent assigned to this Security Token"

**Review by monitz87 — CHANGES_REQUESTED** [review]
There are linting errors

**Comment by monitz87** [comment]
There are 5 code smells as well

**Comment by shuffledex** [comment]
> There are 5 code smells as well

I have to disable the `deprecated` warnings

**Review by monitz87 — APPROVED** [review]

**Pull request merged**